python/foglamp/plugins/south/modbustcp/modbustcp.py

```diff
@@ -175,9 +175,13 @@ def plugin_poll(handle):
                 source_port = int(handle['port']['value'])
             except:
                 raise ValueError
-
-            mbus_client = ModbusTcpClient(host=source_address, port=source_port)
-            _LOGGER.info('Modbus TCP Client is connected: %s, %s:%d', mbus_client.connect(), source_address, source_port)
+            try:
+                mbus_client = ModbusTcpClient(host=source_address, port=source_port)
+            except:
+                _LOGGER.warn('Failed to connect! Modbus TCP host %s on port %d', source_address, source_port)
+                return
+            else:
+                _LOGGER.info('Modbus TCP Client is connected: %s, %s:%d', mbus_client.connect(), source_address, source_port)
 

```

Comment:
Line#203:  modbus_map = json.loads(handle['map']['value']) may be changed to 
 modbus_map = json.loads(handle['map']['value']) if isinstance(handle['map']['value'], str) else handle['map']['value']

Reply:
Nope, We should not. 
Why are you proposing this heck. We should not do just anything to make it work but thoughtfully decide with design!

Reply:
Hack? Real hack has already been merged vide FOGL-2511-B (mutually agreed upon by everyone) and the real "thoughtful design" will be implemented vide FOGL-1827 (a quite long term plan and not currently on radar). So this proposed "hack" immunizes the code from any current or future handling of JSON type configuration items for this plugin.

Reply:
FOGL-2511-B solution was for shim layer. And I was not expecting that we need to put that at plugin level too. is it MUST to put ` if isinstance str` check here along with FOGL-2511-B? is not shim layer only change  will help?

Reply:
Actually, if you can recall, the first comment was made before we applied FOGL-2511-B fix as shim layer fix applies to ALL plugins. But after FOGL-1827 is fixed, we will need to remove json.loads() from all plugins wherever we are using it. Therefore this fix will make code kind of future proof.

Reply:
Yeah

> shim layer fix applies to ALL plugins.

So we can move ahead withOUT any change in plugin for the same. And testing shows results are okay with different changes to json and other other config items/ between restarts etc.? /cc @monika-Dianomic